File: node-service/routes/clearList.js
const express = require('express');
const router = express.Router();
require('dotenv').config();
const { MongoClient } = require('mongodb');

// MongoDB connection URI and database details
const uri = process.env.MONGO_URI;
const client = new MongoClient(uri);

router.delete('/', async (req, res) => {
    try {
        await client.connect();
        const database = client.db("services");
        const collection = database.collection("items");

        // Delete all items
        const result = await collection.deleteMany({});
        res.status(200).json({ message: `${result.deletedCount} items cleared from the list.` });
    } catch (error) {
        res.status(500).json({ message: "Error clearing the list.", error: error.message });
    } finally {
        await client.close();
    }
});

module.exports = router;

File: python-service/routes/clear_list.py
# python-service/routes/clear_list.py
from flask_restful import Resource
from models.database import items_collection

class ClearList(Resource):
    def delete(self):
        result = items_collection.delete_many({})
        return {"message": f"{result.deleted_count} items cleared from the list."}, 200